### PR TITLE
Doc GCS source

### LINKED ISSUE #1734: Document the support of GCS source
### Related code PR

_No response_

### Which part(s) of the docs might be affected or should be updated? And how?

Ingest data - Supported sources -> (New topic) Google Cloud Storage

Syntax is similar to S3_v2, but the credentials are GCS-specific. @wcy-fdu  Please include details here. Thanks! 

```CREATE SOURCE [ IF NOT EXISTS ] source_name 
schema_definition
WITH (
   connector={ 's3' | 's3_v2' |'gcs' },
   connector_parameter='value', ...
)
FORMAT data_format ENCODE data_encode (
   without_header = 'true' | 'false',
   delimiter = 'delimiter'
); ```

### Reference

_No response_

### PR DESCRIPTION
**Pull request by emile-00**
## Info

- **Description**

  - Doc GCS source

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR** 

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/1734

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`CharlieSYH`, `emile-00`, & `hengm3467`).

### CODE REVIEW

**Comment by emile-00** [comment]
@wcy-fdu Please help confirm the accepted data formats and encodes as well as the parameter descriptions. Thank you!

File: docs/ingest/ingest-from-gcs.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+|gcs.credential|Optional. The base64 encoded credential key. If this field is not specified, ADC (application default credentials) will be used. |
+|gcs.service_account|Optional. The service account of the target GCS source. If `gcs.credential` or ADC is not specified, the credentials will be derived from the service account.|
+|match_pattern| Conditional. This field is used to find object keys in `gcs.bucket_name` that match the given pattern. Standard Unix-style [glob](https://en.wikipedia.org/wiki/Glob_(programming)) syntax is supported. |
+
+### Other parameters
+
+|Field|Notes|
+|---|---|
+|*data_format*| Supported data format: `PLAIN`. |
+|*data_encode*| Supported data encodes: `JSON`. |
```
Comment (wcy-fdu) [inline docs/ingest/ingest-from-gcs.md]: GCS source supports both json and csv format.

File: docs/ingest/ingest-from-gcs.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+```
+
+### Connector parameters
+
+|Field|Notes|
+|---|---|
+|gcs.bucket_name |Required. The name of the bucket the data source is stored in. |
+|gcs.credential|Optional. The base64 encoded credential key. If this field is not specified, ADC (application default credentials) will be used. |
+|gcs.service_account|Optional. The service account of the target GCS source. If `gcs.credential` or ADC is not specified, the credentials will be derived from the service account.|
+|match_pattern| Conditional. This field is used to find object keys in `gcs.bucket_name` that match the given pattern. Standard Unix-style [glob](https://en.wikipedia.org/wiki/Glob_(programming)) syntax is supported. |
```
Comment (wcy-fdu) [inline docs/ingest/ingest-from-gcs.md]:
```suggestion
|match_pattern| Conditional. This field is used to find object keys in the bucket that match the given pattern. Standard Unix-style [glob](https://en.wikipedia.org/wiki/Glob_(programming)) syntax is supported. |
```

**Review by wcy-fdu — APPROVED** [review]
Rest LGTM! Thanks!

File: docs/ingest/ingest-from-gcs.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
+---
+id: ingest-from-gcs
+title: Ingest data from Google Cloud Storage
+description: Ingest data from Google Cloud Storage
```
Comment (ShanlanLi) [inline docs/ingest/ingest-from-gcs.md]: We use complete sentences as descriptions, so please add a period^_^.

File: docs/ingest/ingest-from-gcs.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+### Other parameters
+
+|Field|Notes|
+|---|---|
+|*data_format*| Supported data format: `PLAIN`. |
+|*data_encode*| Supported data encodes: `JSON`. |
+|*without_header*| Whether the first line is header. Accepted values: `'true'`, `'false'`. Default: `'true'`.|
+|*delimiter*| How RisingWave splits contents. For `JSON` encode, the delimiter is `\n`. |
+
+## Example
```
Comment (ShanlanLi) [inline docs/ingest/ingest-from-gcs.md]: We use plural forms as long as the heading is countable, even if we just have one example. So pls use plural forms^^.

**Review by ShanlanLi — APPROVED** [review]
Rest LGTM.

File: docs/ingest/ingest-from-gcs.md (hunk 3 above)
Comment (CharlieSYH) [reply docs/ingest/ingest-from-gcs.md]: And provide more info in the description.

File: docs/ingest/ingest-from-gcs.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    primary key(id)
+) 
+WITH (
+    connector = 'gcs',
+    gcs.bucket_name = 'example-bucket',
+    gcs.credential = 'xxxxx'
+) FORMAT PLAIN ENCODE JSON (
+    without_header = 'true',
+    delimiter = '\n'
+);
```
Comment (CharlieSYH) [inline docs/ingest/ingest-from-gcs.md]: It seems like we're missing the ending ``` here.

**Review by CharlieSYH — APPROVED** [review]

**Pull request merged**